Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -730,6 +664,8 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
                      truncateFloatFraction(timeStamp.getTimeStampError() * 1E3, mCollisionPositionCov),
                      collisionTimeMask);
 
+    LOG(INFO) << "vertex.getNContributors() = " << vertex.getNContributors();
+
     auto trackRef = primVer2TRefs[collisionID];
     int start = trackRef.getFirstEntryOfSource(0);
     int ntracks = trackRef.getEntriesOfSource(0);
```

Comment:
In order to account for eventual redundancies among the tracks, I would suggest instead to do
````
using GIDV = o2::dataformats::VtxTrackIndex;
for (int src=GIDV::NSources;src--;) 
{
  int from = trackRef.getFirstEntryOfSource(src), to = from + trackRef.getEntriesOfSource(src);
  for (int it=from;it<to;it++) {
    auto tID = primVerGIs[it];
    if (src == ....) // processing for each type.

  }
}
````
The tID, apart from giving the source and the index within the source, also provides info on the track relation to vertex:
If `tID.isPVContributor()` is `true`, then this track was used in this p.vertex fit and as such, will not be encountered in the the track references of other vertices (even if time-wise it is compatible to them).
If `tID.isAmbiguous()` is `true`, then this track was matched (time-wise) to at least 1 other vertex.
If both flags are false, then the track was matched (time-wise) to given vertex only.

Note that with such a loop when processing e.g. the `src = ITS or TPC`, you will not get those tracks which were contributing to ITS-TPC matches, and in a same fashion, when processing `src = ITS-TPC` you will not get those tracks which were contributing to e.g. `ITS-TPC-TRD` or `ITS-TPC-TOF` or `ITS-TPC-TRD-TOF`.
Therefore, when processing, e.g. `src=ITS-TPC-TRD-TOF`, you can access its contributors on all levels by method e.g.
````
auto contributorsGID = recoCont.getSingleDetectorRefs(tID);
for (int isrc=GIDV::NSources;isrc--;) 
{
  if (contributorsGID[isrc].isIndexSet()) { // this source contributes to this more global track
     // process info of this source.
  }
}
````